builtin/plugin/v5/backend.go

```diff
@@ -0,0 +1,147 @@
+package plugin
+
+import (
+	"context"
+	"net/rpc"
+	"sync"
+
+	uuid "github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/sdk/helper/consts"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/hashicorp/vault/sdk/plugin"
+	bplugin "github.com/hashicorp/vault/sdk/plugin"
+)
+
+// Backend returns an instance of the backend, either as a plugin if external
+// or as a concrete implementation if builtin, casted as logical.Backend.
+func Backend(ctx context.Context, conf *logical.BackendConfig) (logical.Backend, error) {
+	var b backend
+	name := conf.Config["plugin_name"]
+	pluginType, err := consts.ParsePluginType(conf.Config["plugin_type"])
+	if err != nil {
+		return nil, err
+	}
+
+	sys := conf.System
+
+	raw, err := plugin.NewBackendV5(ctx, name, pluginType, sys, conf)
+	if err != nil {
+		return nil, err
+	}
+	b.Backend = raw
+	b.config = conf
+
+	return &b, nil
+}
+
+// backend is a thin wrapper around plugin.BackendPluginClient
```

Comment:
This the comment still accurate here? IIUC it's a wrapper around `BackendPluginClientV5` that could also be just a builtin Backend, depending on what `NewBackendV5` returns.

Reply:
👍